apps/condo/domains/miniapp/utils/b2bAppServiceUserAccess/utils/server.utils.js

```diff
@@ -0,0 +1,216 @@
+        app: {
+            accessRights_some: {
+                accessRightSet: {
+                    [`canManage${pluralize.plural(listKey)}`]: true,
+                    deletedAt: null,
+                },
+                user: { id: user.id, type: 'service', deletedAt: null },
+                deletedAt: null,
+            },
+            deletedAt: null,
+        },
+        status: 'Finished',
+        deletedAt: null,
+    })
+
+    return !isEmpty(B2BAppContexts)
+}
+
+const isServiceUser = ({ authentication: { item: user } }) => {
+    return get(user, 'type') === SERVICE
+}
+
+const getRefSchemaName = (schemaConfig, listKey) => {
+    const pathToOrganizationId = get(schemaConfig, 'pathToOrganizationId', ['organization', 'id'])
+
+    if (!isArray(pathToOrganizationId) || isEmpty(pathToOrganizationId)) {
+        throw new Error('"pathToOrganizationId" must be not empty array!')
+    }
+    for (const pathPart of pathToOrganizationId) {
+        if (!isString(pathPart) || pathPart.trim().length < 1) {
+            throw new Error(`"pathToOrganizationId" must contain array of string! But was: ${pathToOrganizationId}`)
+        }
+    }
+
+    const schema = getSchemaCtx(listKey)
+
+    const schemaFields = get(schema, 'list._fields', {})
+    return get(schemaFields, [pathToOrganizationId[0], 'ref'], null)
+}
+
+/**
+ * Checks that service user can read objects of organization that is connected to linked B2B app
+ *
+ * @param args
+ * @return {Promise<Record<string, any>|false>}
+ */
+const serviceUserCanReadSchemaObjectsIfOrganizationConnectedToLinkedB2BApp = async (args) => {
+    const { listKey } = args
+    if (!isServiceUser(args)) return false
+    const schemaConfig = get(B2B_APP_SERVICE_USER_ACCESS_AVAILABLE_SCHEMAS, listKey)
+    if (!isObject(schemaConfig)) return false
+    const canBeRead = get(schemaConfig, 'canBeRead', true)
+    if (!canBeRead) return false
+    return await canReadByServiceUser(args, schemaConfig)
+}
+
+/**
+ * Checks that service user can manage objects of organization that is connected to linked B2B app
+ *
+ * @param args
+ * @return {Promise<boolean>}
+ */
+const serviceUserCanManageSchemaObjectsIfOrganizationConnectedToLinkedB2BApp = async (args) => {
+    const { listKey } = args
+    if (!isServiceUser(args)) return false
+    const schemaConfig = get(B2B_APP_SERVICE_USER_ACCESS_AVAILABLE_SCHEMAS, listKey)
+    if (!isObject(schemaConfig)) return false
+    const canBeManage = get(schemaConfig, 'canBeManage', true)
+    if (!canBeManage) return false
+    const refSchemaName = getRefSchemaName(schemaConfig, listKey)
+    return await canManageByServiceUser(args, schemaConfig, refSchemaName)
+}
+
+/**
+ * Merge two access filters or return false
+ *
+ * @param a {any}
+ * @param b {any}
+ * @return {Record<string, any>|{OR: [{AND: Record<string, any>[]},{AND: Record<string, any>[]}]}||false}
+ */
+const mergeAccessFilters = (a, b) => {
```

Comment:
Why we need some OR's here?

Reply:
OR will double db-table scan. 

Can we just figure out once, that it's a service user. and if so, return his access.
Else return staff / resident access

Reply:
Currently, the plugin combines two queries - default filter + filter by miniapp.

If accesses are not explicitly set for a service account, then it uses as for employee. And if you change the filters everywhere for the service user, then I’m not sure that nothing will break in miniapps/for partners/contractors/integrations

Reply:
We can assume, that all users, linked to apps are of `SERVICE` type

Reply:
removed this function